### PR TITLE
doctor refactored

### PR DESCRIPTION
**Pull request by eabazi**
1. Remove some commented out code (ts and html) 
2. Add reference to room as doctor office 
3. Edit title from all lower case 
4. tab -> 2 spaces EVERYWHERE 
5. Make parameter of addDoctor a doctor object 
6. Try adding doctor object to collection 
7. dr -> doctor 
8. Make a provider and move all non-gui things there 
9. Separated providers (room, speciality)

### CODE REVIEW

**Review by jHelsing — APPROVED** [review]
Looks good!

File: src/providers/doctor/doctor.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  private doctorObservable: Observable<Doctor[]>;
+
+  constructor(public fireStore: AngularFirestore, public alertCtrl: AlertController) {
+    this.doctorCollection = fireStore.collection<Doctor>('/doctors');
+    this.doctorObservable = this.doctorCollection.snapshotChanges().map((actions) => actions.map((action) => ({
+      $id: action.payload.doc.id, ...action.payload.doc.data() as Doctor,
+    })));
+  }
+
+  public getDoctor() {
```
Comment (ckarlss0n) [inline src/providers/doctor/doctor.ts]: **Suggestion:** Change name to `getDoctors` as it returns multiple doctors.

Comment (oskarlignell) [reply src/providers/doctor/doctor.ts]: Fixed.

File: src/pages/add-doctor/add-doctor.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+})
+
+export class DoctorRegistration {
+  public doctor = {};
+  private doctorObservable: Observable<Doctor[]>;
+  private roomObservable: Observable<Room[]>;
+  private specialityObservable: Observable<Speciality[]>;
+
+  constructor(public navCtrl: NavController, public alertCtrl: AlertController, public doctorProvider: DoctorProvider, public roomProvider: RoomProvider, specialityProvider: SpecialityProvider) {
+    this.doctorObservable = doctorProvider.getDoctor();
```
Comment (davidemdot) [inline src/pages/add-doctor/add-doctor.ts]: I think it should be getDoctor**s**() here.

**Review by davidemdot — APPROVED** [review]

**Pull request merged**